test/Constraints/closures.swift

```diff
@@ -299,7 +299,7 @@ struct Thing {
   init?() {}
 }
 // This throws a compiler error
-let things = Thing().map { thing in  // expected-error {{unable to infer complex closure return type; add explicit type to disambiguate}} {{34-34=-> (Thing) }}
+let things = Thing().map { thing in  // expected-error {{unable to infer complex closure return type; add explicit type to disambiguate}} {{34-34=-> Thing }}
```

Comment:
We obviously don't want the parens printed here (or in the other tests that changed), but it would be good to confirm that we know why they are no longer there, and that we aren't inadvertently removing parens that might be wrapping a tuple somewhere where it makes a semantic difference in the current implementation.

Reply:
Yes, I double-checked that we are actually not loosing parens in other places when I first saw that this test has changed, the difference comes from the fact that we'd generate a type from parameters now which only has parens in places that require them, otherwise everything is represented by the type itself.